### PR TITLE
OpenID

### PR DESCRIPTION
**Pull request by lelemm**
Cleaning up the PR #381

### CODE REVIEW

**Comment by lelemm** [comment]
@MatissJanis so, I changed all the review points you pointed out on the old PR, but as a side effect, some stuff needs to change on the UI side. I changed almost everything, but I will need to refactor the user access page.
I will take a look at it tomorrow and then I tell you to review again, ok?

**Comment by lelemm** [comment]
You can review again @MatissJanis

File: src/app-secrets.js
Comment (lelemm) [reply src/app-secrets.js:34]: removing the return would let the code keep executing. I guess I could remove the "null" from the return, but not the return itself

File: src/scripts/reset-password.js
Comment (lelemm) [reply src/scripts/reset-password.js:11]: done

File: src/app-secrets.js
Comment (lelemm) [reply src/app-secrets.js:34]: done

File: src/app-admin.test.js
Comment (lelemm) [reply src/app-admin.test.js:66]: can you create a Committable suggestion for this?

File: src/app-admin.test.js
Comment (lelemm) [reply src/app-admin.test.js:28]: all calls from getAccountDb are NOT async, what is the reasoning behind this?

Comment (lelemm) [reply src/app-admin.test.js:28]: ok

File: src/app-admin.test.js
Comment (lelemm) [reply src/app-admin.test.js:66]: as said in another point, database calls are not async

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:15]: done

File: src/app-admin.test.js
Comment (lelemm) [reply src/app-admin.test.js:66]: ok

File: migrations/1718889148000-openid.js
Comment (lelemm) [reply migrations/1718889148000-openid.js:13]: the value is already encrypted, just moving from one structure to another

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:173]: like?

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:193]: This is the expected flow. If the server is compromised before hand, the attacker can already attack

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:241]: I'm open for a commitable suggestion

File: src/accounts/password.js
Comment (lelemm) [reply src/accounts/password.js:68]: this is how the system works today without openid. without openid there is no multuser, doesn't make a difference to have a name and username, they are never used or displayed

File: src/accounts/password.js
Comment (lelemm) [reply src/accounts/password.js:79]: same as last comment

File: src/accounts/password.js
Comment (lelemm) [reply src/accounts/password.js:120]: this is done in the expressjs route

File: src/app-account.js
Comment (lelemm) [reply src/app-account.js:43]: the bootstrap() method is saniting it, isn't?

File: src/accounts/password.js
Comment (lelemm) [reply src/accounts/password.js:24]: no, the update without where is the expected change. when we bootstrap with password, that will be the only enabled auth method, so we disable all the rest

File: src/app-openid.js
Comment (lelemm) [reply src/app-openid.js:18]: all database calls are sync when using better-sqlite3, so, there is no async calls needed

File: src/app-openid.js
Comment (lelemm) [reply src/app-openid.js:85]: add a console log of the message if the error is not being send to the client

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:164]: none of the calls are async

File: src/app-account.js
Comment (lelemm) [reply src/app-account.js:134]: its needed

File: src/app-openid.js
Comment (lelemm) [reply src/app-openid.js:83]: give example of how to implement isValidRedirectUrl, at the table 'auth', when method = 'openid', extra_data has a server_hostname that is the current domain of the application

File: src/app-openid.js
Comment (lelemm) [reply src/app-openid.js:27]: the openid lib validates it and there is a basic validation at enableOpenID

File: src/app-openid.js
Comment (lelemm) [reply src/app-openid.js:83]: not sure if this will work:
```
if (redirectUrl.hostname === serverHostname) {
```
because the serverhostname must be (protocol)://hostname (example: https://google.com)

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:254]: can't, some routes can't use this as they can accept anonymous

File: src/app-sync.js
Comment (lelemm) [reply src/app-sync.js:117]: removed this if

File: src/load-config.js
Comment (lelemm) [reply src/load-config.js:81]: that's all true, but currenctly the app supports non expiring token, so I was asked to maintain it

File: src/app-sync/services/files-service.js
Comment (lelemm) [reply src/app-sync/services/files-service.js:163]: this isn't needed since this will be always called after a file find

File: src/app-sync/services/files-service.js
Comment (lelemm) [reply src/app-sync/services/files-service.js:106]: fixed

File: jest.global-setup.js
Comment (lelemm) [reply jest.global-setup.js:79]: done

File: src/account-db.js
Comment (lelemm) [reply src/account-db.js:151]: done

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js]: already done

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:67]: it can accept http too

File: src/app-secrets.js
Comment (lelemm) [reply src/app-secrets.js:45]: not changing to this. too confuse

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:229]: this return will not work since its inside a anonymous method so the return will be lost

**Review by MatissJanis — APPROVED** [review]
LGTM on my end. Just need another maintainer to review & approve

**Review by joel-jeremy — APPROVED** [review]
LGTM

File: src/app-sync/tests/services/files-service.test.js
Comment (jfdoming) [reply src/app-sync/tests/services/files-service.test.js:160]: I think I agree with the bot here, mind adding some tests for this?

File: src/load-config.js
Comment (jfdoming) [inline src/load-config.js:56]: Curious why this was removed?

File: src/app-sync/services/files-service.js
Comment (jfdoming) [inline src/app-sync/services/files-service.js:125]: This code as written seems a little dangerous to me. In particular, it's used in `app-sync` which just passes in `res.locals.user_id`—right now, it may be the case that this is always non-null, but I can imagine a future PR which accidentally makes that not the case. Do we need the `null` fallback to admin for something? If not, can we remove it and error loudly if the user ID is null?

File: src/app-sync.js
Comment (jfdoming) [inline src/app-sync.js:330]: It may not be for this PR, but this seems a little inefficient to me. I think it would be better to have `findUsersWithAccess(rows.map(({id}) => id))` so we don't have to look at disk for each separate file. What do you think?

(same comment below for `/get-user-file-info`)

**Review by jfdoming — CHANGES_REQUESTED** [review]
Left only a couple of minor comments, thanks for pushing through this! I'm requesting changes only so we don't accidentally merge this before we can discuss the comments, but it does look broadly good to me 💯

File: src/app-sync/services/files-service.js
Comment (lelemm) [reply src/app-sync/services/files-service.js:125]: Its nullable because it was messing with the test cases. Fixed the test cases and removed the null

File: src/load-config.js
Comment (lelemm) [reply src/load-config.js:56]: it was removed (I don't remember why), but I added it back on the end of the defaultConfig object. I moved it back to the top

File: src/app-sync/tests/services/files-service.test.js
Comment (lelemm) [reply src/app-sync/tests/services/files-service.test.js]: added a test for this

**Review by MatissJanis — APPROVED** [review]

**Review by jfdoming — COMMENTED** [review]
Hey! I did another round of review and found a few more things, sorry for the delay and back-and-forth 😞 Most of these can be punted but I think it'd be good to either address or discuss the issue with 404s/403s

File: src/app-openid.js
Comment (jfdoming) [reply src/app-openid.js:59]: Sure! I'll make a ticket to track a fix for this. We should make sure to address this before we start advertising this feature, though.

File: src/app-sync.js
Comment (jfdoming) [reply src/app-sync.js:330]: Yeah, it makes sense. I'll make a follow-up issue to track this as an optimization

File: src/account-db.js
Comment (lelemm) [reply src/account-db.js:133]: we used to have the possibility to enable openid from the config file. Since we changed the flow to need the user to bootstrap with password once, then enable openid, this was removed, but I forgot to remove this parameter

File: src/services/user-service.js
Comment (lelemm) [reply src/services/user-service.js:61]: this method is not used anymore. removed

File: src/load-config.js
Comment (lelemm) [reply src/load-config.js:80]: changed to check openid enabled

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:266]: changed to 10 minutes fallback

File: src/accounts/openid.js
Comment (lelemm) [reply src/accounts/openid.js:110]: ok

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:192]: I changed the code to check if the user has access to this file, if not send it a forbidden message. This way a bad actor will not even have a response from the file not existing in first place

Comment (lelemm) [reply src/app-admin.js:208]: done

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:231]: done

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:285]: done

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:330]: done

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:366]: done

**Comment by lelemm** [comment]
I changed all to 404 but add one missing forbidden on the get /access

File: src/app-account.js
Comment (lelemm) [reply src/app-account.js:49]: I don't want to leak errors

File: src/app-admin.js
Comment (lelemm) [reply src/app-admin.js:231]: the 404 is behind a forbidden 'wall', if the bad actor is a admin or a file owner, there is no leakage since they can already have access to file list

**Review by jfdoming — APPROVED** [review]
Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock

**Comment by lelemm** [comment]
don't approve this yet. found some inconsistencies

**Comment by lelemm** [comment]
there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good

**Comment by lelemm** [comment]
> Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock

only admins or file owners can call this route.
- admins for all files
- file owners for their own files

**Comment by jfdoming** [comment]
> > Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock
> 
> only admins or file owners can call this route.
> 
> * admins for all files
> * file owners for their own files

Ah, I see—this one always returns 403 instead of always returning 404. Makes sense

**Comment by jfdoming** [comment]
> there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good
Oof, that is unfortunate

**Review by jfdoming — APPROVED** [review]

**Comment by lelemm** [comment]
> > there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good
> 
> Oof, that is unfortunate

Already fixed.
I'm working on the PR for this on the other repo

**Review by MatissJanis — APPROVED** [review]
Thanks!

**Pull request merged**
